bin.js

```diff
@@ -46,8 +48,13 @@ log.info('begin', 'Prebuild version', prebuildVersion)
 delete process.env.NVM_IOJS_ORG_MIRROR
 delete process.env.NVM_NODEJS_ORG_MIRROR
 
+var execPath = process.env.npm_execpath || process.env.NPM_CLI_JS
+
 if (rc.install) {
-  if (!(typeof pkg._from === 'string')) {
+  if (util.isYarnPath(execPath) && /node_modules/.test(home())) {
```

Comment:
I just tested `v5.0.2` and it still compiles from source.

Stupid me I used `process.cwd()` for testing and added `home()` here. I'll make another PR. Sorry for the noise.